packages/react/dropdown-menu/src/DropdownMenu.tsx

```diff
@@ -266,38 +267,26 @@ const DropdownMenuRootContent = React.forwardRef<
       {...contentProps}
       ref={forwardedRef}
       portalled={portalled}
-      onCloseAutoFocus={(event) => {
-        props.onCloseAutoFocus?.(event);
-
-        if (!event.defaultPrevented) {
-          if (!hasInteractedOutsideRef.current) context.triggerRef.current?.focus();
-          // Always prevent auto focus because we either focus manually or want user agent focus
+      onCloseAutoFocus={composeEventHandlers(props.onCloseAutoFocus, (event) => {
+        if (context.modal) context.triggerRef.current?.focus();
+        // Always prevent auto focus because we either focus manually or want user agent focus
+        event.preventDefault();
+      })}
+      onPointerDownOutside={composeEventHandlers(props.onPointerDownOutside, (event) => {
+        const originalEvent = event.detail.originalEvent as PointerEvent;
+        const ctrlLeftClick = originalEvent.button === 0 && originalEvent.ctrlKey === true;
+        const isRightClick = originalEvent.button === 2 || ctrlLeftClick;
+        if (context.modal && isRightClick) {
+          // close after the browser context menu handler so that
+          // context menu options are inert (for the body)
+          document.addEventListener(
+            'contextmenu',
+            () => window.setTimeout(() => context.onOpenChange(false), 0),
+            { once: true }
```

Comment:
I know we have added a `document.removeListener` in a cleanup effect when we do this in the past but I don't think it's needed here because it'll all happen as part of the same event (the right click will instantly trigger context menu event).